kubernetes/icinga2/files/scripts/election.py

```
@@ -0,0 +1,236 @@
+API_URL        = "https://localhost:5665"
+API_USER       = os.environ.get("ICINGA_API_USER", "icingaAdmin")
+API_PASS       = os.environ.get("ICINGA_API_PASSWORD", "icingaAdmin")
+
+
+def run_promote():
+    """Called when this pod acquires the Lease and becomes leader.
+
+    Idempotent: safe to call multiple times. Verifies content availability
+    before symlink creation to avoid promoting with empty config.
+    """
+    log.info(f"=== PROMOTE: {POD_NAME} becoming leader ===")
```

Ruff (0.16.0)

[warning] Logging statement uses f-string

(G004)